### PR TITLE
👷 Add a GitHub workflow to prevent blocked PRs from being merged

### PR DESCRIPTION
**Pull request by ebouchut**
## Description

This PR adds a new feature to PRs in this GitHub repository.
A PR can depend on:
- **another PR** from the same repository
- a GitHub issue from the same repository

To indicate that PR2 depends on PR1,
add **`Blocked by #1`** to PR2's description

This PR adds a new GitHub workflow (`.github/workflows/pr-dependencies-check.yml`) using the [Levi-Lesches/blocking-issues](https://github.com/Levi-Lesches/blocking-issues) GitHub Action. 
It runs each time a PR is created, updated, closed, or re-opened.
It checks if the PR is blocked. 
If so it fails, preventing the PR from being merged. 

⏱️ The workflow takes ~10 seconds to run.

### PR Dependent on another PR

✅ This works fine provided both PRs are in the same repository.
Cross-repository PR dependencies are not supported.

### PR Dependent on an Issue 

~~I experimented with making a PR dependent on a **GitHub Issue**.
Based on my testing, it does not work as closing this issue does not trigger the workflow. 
Therefore this dependency change is not taken into account automatically. 
I needed to edit the PR description to trigger the workflow again which is not optimal!  
❌ This is why, I do not recommend making a PR dependent on a GH issue.~~

**EDIT:** ✅ It now works fine. I made a mistake and the indentation was off. Now fixed.  

## Resources

- `Levi-Lesches/blocking-issues` GitHub Action: https://github.com/Levi-Lesches/blocking-issues
- Discussion about the need for such a feature in GitHub and the open-source workarounds in the meantime: https://github.com/orgs/community/discussions/4477#discussioncomment-1039627

### CODE REVIEW

**Comment by ebouchut** [comment]
This PR adds a GitHub Workflow to check for PRs blocked by another PR and/or issue.
It prevents the dependent PR from being merged while the blocking  PR/issue is opened.

GitHub does not provide a way to mark a PR dependent on another PR.
For example, making PR-2 dependent on PR-1.
- PR-1 is the parent PR
- PR-2 is the child PR


While searching for open-source alternatives to fill this void,
I came across an interesting discussion on GitHub itself.
It was started a few years ago by @isaacs on his isaacs/github repo.
GitHub subsequently hired him, and the repo morphed into this discussion (now hosted on Github).


The need is twofold:

1. Be able to compare the difference between PR-1 (parent) and PR-2 (child).
2. Prevent PR-2 from being merged before PR-1 is merged

I have paraphrased above what is explained in this message: 
  https://github.com/orgs/community/discussions/4477#discussioncomment-1293255

A few tools answer point 2 (i.e. avoid an incorrect merge of the child PR before the parent PR):
  https://github.com/orgs/community/discussions/4477#discussioncomment-1039627


I tried every one of them and decided to go with the `blocking-issues` Github Action.
  https://github.com/Levi-Lesches/blocking-issues

> A GitHub action to **label issues and PRs when they're blocked by another issue or PR**.

This PR adds a GitHub Workflow to check for PRs and GitHub issues blocked by another PR and/or Issue.
For instance, we can edit the description of PR-2 and mark it as blocked by PR-1 like this:
```txt
Blocked by #1
```
The workflow will now prevent PR-2 from being merged before PR-1 is merged.

**Review by MikePlante1 — APPROVED** [review]
LGTM

**Comment by ebouchut** [comment]
Thanks @MikePlante1 .

❌ The [workflow failed](https://github.com/nightscout/trio-docs/actions/runs/11653740360/job/32446461215#step:2:50) because it could not create the `blocked` label.
Could you create it by hand [here](https://github.com/nightscout/trio-docs/labels) like this and edit the PR description to re-trigger the checks:

> CleanShot 2024-11-03 at 19 40 49@2x

**Comment by ebouchut** [comment]
ℹ️ For the record, the `Checks` of a PR may display this error 
> Error: Resource not accessible by integration

as long as it is in **Draft** mode.

To display the detailed the _Checks_ logs, click the **`Details`** link next to
> Prevent Blocked PRs from being Merged / blocking_issues (pull_request)

**Pull request merged**
